experimental/cuda2/pending_queue_actions.c

```diff
@@ -525,20 +676,108 @@ iree_status_t iree_hal_cuda2_pending_queue_actions_issue(
 
   iree_slim_mutex_unlock(&actions->action_mutex);
 
-  // Now go through the ready list and issue the actions to the GPU.
-  for (iree_hal_cuda2_queue_action_t* action = ready_list.head;
-       action != NULL;) {
-    iree_hal_cuda2_queue_action_t* next_action = action->next;
-    action->next = NULL;
+  iree_hal_cuda2_atomic_slist_entry_t* entry = NULL;
+  // TODO: avoid host allocator malloc; use some pool for the allocation.
+  IREE_RETURN_AND_END_ZONE_IF_ERROR(
+      z0, iree_allocator_malloc(actions->host_allocator, sizeof(*entry),
+                                (void**)&entry));
+  entry->ready_list_head = ready_list.head;
+
+  // Now push the ready list to the worker and have it to issue the actions to
+  // the GPU.
+  iree_hal_cuda2_working_area_t* working_area = &actions->working_area;
+  iree_hal_cuda2_ready_action_slist_push(&working_area->ready_worklist, entry);
+  iree_atomic_store_int32(&working_area->working_state,
+                          IREE_HAL_CUDA_WORKER_STATE_WORKLOAD_PENDING,
+                          iree_memory_order_relaxed);
+  iree_notification_post(&working_area->notification, IREE_ALL_WAITERS);
 
-    IREE_RETURN_AND_END_ZONE_IF_ERROR(
-        z0, iree_hal_cuda2_pending_queue_actions_issue_execution(action));
-    action->events = NULL;
-    action->event_count = 0;
+  IREE_TRACE_ZONE_END(z0);
+  return iree_ok_status();
+}
 
-    action = next_action;
+//===----------------------------------------------------------------------===//
+// Worker routines
+//===----------------------------------------------------------------------===//
+
+static bool iree_hal_cuda2_worker_has_incoming_request(void* args) {
+  iree_hal_cuda2_working_area_t* working_area =
+      (iree_hal_cuda2_working_area_t*)args;
+  int value = iree_atomic_load_int32(&working_area->working_state,
```

Comment:
int32_t (here and elsewhere) - avoid ints and such especially when using typed things